### PR TITLE
#163518702 User can report article

### PR DESCRIPTION
**Pull request by niomwungeri-fabrice**
#### What does this PR do?
Report an article
#### Description of Task to be completed?
Have this endpoints work
EndPoint:
* `GET /article/:slug/report` reporting an article
#### How should this be manually tested?
* Fetch the branch `ft-user-can-report-article-163518702`
* Checkout the branch `ft-user-can-report-article-163518702`
* Setup your database. 
* Run tests `yarn test`
* Start the server `yarn start`
* Create a user `POST: /api/v1/users` by providing `email, password, and username`
* Login a user `POST: /api/v1/auth/login` by providing `email and password` you will get a token
* Create an article `POST: /api/v1/article` by providing `title, description, body, tagsList: []` 
* Report an article `PUT: /api//v1/:slug/report` by providing in body `description`(mandatory)
* Pass token into the header and try to access one of the above endpoints in PR description using `POSTMAN`
#### What are the relevant pivotal tracker stories?
163518702

### CODE REVIEW

**Review by espoirMur — CHANGES_REQUESTED** [review]
- Rename the database table name
- Create a separate file for your test for maintainability

File: sample.env [hunk 1]
```diff
@@ -3,7 +3,7 @@ DB_PASSWORD = 'user_password';
 DB_NAME ='development_database_name';  
 DB_HOST='localhost';
 TEST_DB_NAME ='test_database_name';  
-SECRET_OR_KEY ='your_jwt_key';  
+SECRET_KEY ='your_jwt_key';  
 BCRYPT_HASH_ROUNDS = 'number, better do greater than 10'
 SENDGRID_API_KEY = '********************'
 SERVER_HOST = '***yourhost***/ap/v1'
```
Comment (espoirMur) [inline sample.env]: @niomwungeri-fabrice, should be  `***yourhost***/api/v1`

File: tests/integration/article.test.js [hunk 2]
```diff
@@ -38,6 +50,8 @@ describe("# Articles endpoints", () => {
         .set("Authorization", `Bearer ${token}`)
         .send(newArticle)
         .end((err, res) => {
+          createdArticle = res.body;
+          validArticleId = res.body.article.id;
```
Comment (espoirMur) [inline tests/integration/article.test.js]: @niomwungeri-fabrice , you can use destructuring here

File: controllers/articlesController.js [hunk 3]
```diff
@@ -53,7 +64,88 @@ export default class PostController {
...
+      const { id: userId } = req.user;
+      const { slug } = req.params;
+      const user = await User.findOne({ where: { id: userId } });
+      const article = await Article.findOne({ where: { slug } });
+      if (!article || !user)
+        return res.status(NOT_FOUND).json({
+          status: NOT_FOUND,
+          message: `The article with this slug ${slug} doesn't exist`
+        });
+      const { id: articleId } = article.dataValues;
```
Comment (Musigwa-zz) [inline controllers/articlesController.js]: Please consider referring a specific article with its `slug` instead of `articleId` as found in README.md file.

**Review by Musigwa-zz — CHANGES_REQUESTED** [review]
Please consider using **`slug`** instead of **`articleId`** in every file that searches for a specific article. Good work indeed.

**Review by espoirMur — APPROVED** [review]

File: controllers/articlesController.js [hunk 4]
```diff
@@ -269,4 +276,39 @@ export default class PostController {
...
+      const { id: userId } = req.user;
+      const { description } = req.body;
+      
+      if (!description || undefined) {
+        return res.status(BAD_REQUEST).json({ message: "Please, give a reason" });
+      }
+      const article = await Article.findOne({
+        where: { slug }
+      });
+      const { articleId } = article.dataValues;
```
Comment (Musigwa-zz) [inline controllers/articlesController.js:298]: Please use the `slug` instead of `articleId`

File: controllers/articlesController.js [hunk 5]
```diff
@@ -269,4 +276,39 @@ export default class PostController {
...
+      const reportArticle = await ReportArticle.create({
+        articleId,
```
Comment (Musigwa-zz) [inline controllers/articlesController.js:300]: Please use the `slug` and make sure to change it everywhere.

File: controllers/auth/loginController.js [hunk 6]
```diff
@@ -13,6 +13,15 @@ const { User } = models;
  * @class
  */
 class Login {
+  /**
+   *
+   * SignIn.
```
Comment (Musigwa-zz) [inline controllers/auth/loginController.js:18]: Please check the conventions of comments from `articleController.js` file. Look at how it is documented as in some of the other files of this project.

File: tests/integration/article.test.js [hunk 7]
```diff
@@ -133,6 +135,74 @@ describe("# Articles endpoints", () => {
...
+        .put(`/api/v1/articles/${validSlug}/report`)
+        .set("Authorization", `Bearer ${token}`)
+        .send({ description: "abusive" })
+        .end((err, res) => {
+          expect(res.status).equals(OK);
+          expect(res.body.message).to.contain("Article reported");
+          done();
+        });
+    });
+    it("should should ask for description", done => {
```
Comment (Musigwa-zz) [inline tests/integration/article.test.js:163]: Check to make sure if this case scenario is a duplicate.

File: tests/integration/article.test.js [hunk 8]
```diff
@@ -133,6 +135,74 @@ describe("# Articles endpoints", () => {
...
+        .put(`/api/v1/articles/${validSlug}/report`)
+        .set("Authorization", `Bearer ${token}`)
+        .send({ description: "" })
+        .end((err, res) => {
+          expect(res.status).equals(BAD_REQUEST);
+          expect(res.body.message).to.contain("Please, give a reason");
+          done();
+        });
+    });
+    it("should deny the request if no access-token provided", done => {
```
Comment (Musigwa-zz) [inline tests/integration/article.test.js]: This might be a duplicate

**Review by Musigwa-zz — CHANGES_REQUESTED** [review]
Please consider refining the quoted points. Good work indeed.

File: tests/integration/article.test.js (hunk 2 above)
Comment (niomwungeri-fabrice) [reply tests/integration/article.test.js]: If you want a different variable name compared to what the object has, it not necessary to destructure an object. right? @espoirMur

File: tests/integration/article.test.js (hunk 7 above)
Comment (niomwungeri-fabrice) [reply tests/integration/article.test.js]: It was, thanks

File: controllers/articlesController.js [hunk 9]
```diff
@@ -269,4 +276,39 @@ export default class PostController {
...
+   * @returns {Object} - It returns the request response object.
+   */
+
+  static async reportArticle(req, res) {
+    try {
+      const { slug } = req.params;
+      const { id: userId } = req.user;
+      const { description } = req.body;
+      
+      if (!description || undefined) {
```
Comment (Musigwa-zz) [inline controllers/articlesController.js]: if(!description) checks whether the value is `null || undefined`. Please remove that `undefined`

**Review by Musigwa-zz — APPROVED** [review]

**Pull request merged**